core/sdk-core/src/main/java/software/amazon/awssdk/core/RequestOverrideConfiguration.java

```diff
@@ -55,6 +56,7 @@ public abstract class RequestOverrideConfiguration {
     private final EndpointProvider endpointProvider;
     private final CompressionConfiguration compressionConfiguration;
     private final List<SdkPlugin> plugins;
+    private final List<ProgressListener> progressListeners;
```

Comment:
Should we just have ProgressListener and then have ProgressListenerListenerChain something similar to [SdkTokenProviderChain](https://github.com/aws/aws-sdk-java-v2/blob/9cf5f8adda439c4ed1ac7ae0dd68d71931090549/core/auth/src/main/java/software/amazon/awssdk/auth/token/credentials/SdkTokenProviderChain.java#L51) ?

Reply:
hmm.. good point we could do it this way too, but do we see any advantages either way ? V1 does it this way too. But under the hood, both approaches are similar in that we maintain a list of listeners. We are also staying consistent with TransferListeners by maintaining a list of ProgressListeners. The chain is a wrapper around this list, in essence. We could think of taking a detour from TransferListener's way of implementation if we see definite advantages to the approach.

Reply:
Lets discuss this in SurfaceAPI review.
Could please add a TODO : comment here saying will 
```
// TODO : need to discuss this in SurfaceAPI review
```

Reply:
done

Reply:
Discussed offline with Matt. Our guideline around using a Chain abstraction, i.e. V1's progress listener or the SdkTokeProvider, where we update one entity. In this case we have multiple listener methods to be invoked and we maintain a list, as done by plugins https://github.com/aws/aws-sdk-java-v2/blob/5c2868f4512f936000cf084f69bb0707e029a71e/core/sdk-core/src/main/java/software/amazon/awssdk/core/RequestOverrideConfiguration.java#L70